src/Classes/ServiceAPI/MyRadio_Event.php

```diff
@@ -252,4 +380,17 @@ public function toDataSource($mixins = [])
             'master' => (is_null($this->masterId) ? null : self::getInstance($this->masterId)->toDataSource($mixins))
         ];
     }
+
+    public function canWeEdit()
+    {
+        return $this->getHost()->getID() !== MyRadio_User::getCurrentOrSystemUser()->getID()
```

Comment:
Surely this should be EQUALS not NOT EQUAL as if host==current, they can edit

Reply:
oops